### PR TITLE
refactor(navbar): define design tokens for colors

### PR DESCRIPTION
**Pull request by emmenko**
image image

### CODE REVIEW

File: packages/application-shell/src/components/navbar/navbar.mod.css [hunk 1]
```diff
@@ -70,15 +73,24 @@
...
   height: var(--icon-size);
 }
+.list-item .icon > svg *:not([fill='none']) {
+  /* stylelint-disable-next-line csstools/value-no-unknown-custom-properties */
+  fill: var(--color-for-navbar-icon);
+}
+.list-item .icon__active > svg *:not([fill='none']) {
+  /* stylelint-disable-next-line csstools/value-no-unknown-custom-properties */
+  fill: var(--color-for-navbar-icon-when-active);
+}
```
Comment (emmenko) [inline packages/application-shell/src/components/navbar/navbar.mod.css:83]: We're overwriting the fill colors directly here (similarly to what the icon itself does internally) to be able to use the design tokens, even though the icon component has a color prop.

File: packages/application-shell/src/components/navbar/navbar.tsx [hunk 2]
```diff
@@ -702,6 +699,20 @@ const NavBar = <AdditionalEnvironmentProperties extends {}>(
...
+              height: 32px;
+              width: 100%;
+              background: linear-gradient(
+                0deg,
+                ${appkitDesignTokens.backgroundColorForNavbar} 0%,
+                rgba(0, 0, 0, 0) 100%
+              );
+              z-index: 1;
+            `}
+          />
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/components/navbar/navbar.tsx:715]: **nit**: perhaps we could have a local styled component for this. Perhaps something like `MenuItemContainer` or `MenuItemWraper`

File: packages/application-components/src/theming.ts [hunk 3]
```diff
@@ -4,18 +4,33 @@ import {
 } from '@commercetools-uikit/design-system';
 
 const appKitSpacing55 = '40px';
+// TODO: move to uikit?
```
Comment (CarlosCortizasCT) [inline packages/application-components/src/theming.ts:7]: I also agree 👍

File: packages/application-shell/src/components/navbar/navbar.mod.css [hunk 4]
```diff
@@ -21,7 +21,8 @@
   flex-grow: 1;
   width: var(--width-left-navigation);
   z-index: 19999;
-  background: var(--color-accent);
+  /* stylelint-disable-next-line csstools/value-no-unknown-custom-properties */
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/components/navbar/navbar.mod.css:24]: Perhaps we can update the [stylelint config](https://github.com/commercetools/merchant-center-application-kit/blob/main/stylelint.config.js#L39) to include the tokens from the `theming.ts` file? (although we would need to read a typescript file from a javascript one)

**Review by CarlosCortizasCT — APPROVED** [review]

File: packages/application-shell/src/components/navbar/navbar.mod.css (hunk 4 above)
Comment (emmenko) [reply packages/application-shell/src/components/navbar/navbar.mod.css:24]: I thought about it but I'm not sure it's worth the effort, as we either need to maintain the list manually or generate some kind of JSON file.

Keep in mind that when we refactor the navbar, we'll be using CSS-in-JS, so the stylelint config (for the appkit repo) is not needed anymore.

File: packages/application-shell/src/components/navbar/navbar.tsx (hunk 2 above)
Comment (emmenko) [reply packages/application-shell/src/components/navbar/navbar.tsx]: See https://github.com/commercetools/merchant-center-application-kit/pull/2897/commits/996fb1f9c79809c5dd585bf4345c0032d2b47a8b

**Review by kark — APPROVED** [review]

**Pull request merged**